### PR TITLE
docs: add 4.0.0 release notes

### PR DESCRIPTION
**Pull request by ckerr**
rendered text: 

https://github.com/transmission/transmission/blob/6d238ec98c58d0fa88140368b1d390717ddc8279/news/news-4.0.0.md

### CODE REVIEW

File: news/news-4.0.0.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+## Highlights
+
+This is a major release, both in numbering and in effort! It's been in active development for over
+a year and has a _huge_ list of changes -- over a thousand commits -- since Transmission 3.00.
+Some of the highlights include:
+
+- [**Resource Efficiency**](#resource-efficiency) - Use less memory and fewer CPU cycles
+- [**Better Community**](#community) - Pull requests welcomed and used
+- [**Code Modernization**](#code-modernization) - Migrated from C90 to modern C++
+- [**New Features**](#newfFeatures) - What would a major release be without them?
```
Comment (Coeur) [inline news/news-4.0.0.md:12]: typo: (#newfFeatures) -> (#new-features)

File: news/news-4.0.0.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+### Community
+
+- The project is much more responsive to bug reports and code submissions than it has been in the past.
+- There is a new group of volunteer contributors who are working on Transmission!
+- Transmission 4.0.0 includes over 350 new community commits (see [the Thank You section below](#thank-you) and in the previous 4.0.0 betas) since 3.00 and welcomes new contributors.
+- Documentation has been moved into the `transmission/transmission` so that contributors can submit PRs to improve it.
+
+### Code Modernization
+
+- The entire codebase has been migrated from C to C++. In the process, we've removed thousands of lines of custom code and used standard C++ tools instead. The core's code has shrunk by 18%. The core codebase has extensively refactored to be more testable and maintainable.
```
Comment (mikedld) [inline news/news-4.0.0.md:30]:
```suggestion
- The entire codebase has been migrated from C to C++. In the process, we've removed thousands of lines of custom code and used standard C++ tools instead. The core's code has shrunk by 18%. The core codebase has been extensively refactored to be more testable and maintainable.
```

File: news/news-4.0.0.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+* Fixed bug that caused magnet links to always be paused when added. ([#4528](https://github.com/transmission/transmission/pull/4528))
+* Fixed `4.0.0-beta.1` regression that broke the Piece View "blinking" when a piece completes. ([#4587](https://github.com/transmission/transmission/pull/4587))
+* Ensured that the preferences window is centered the first time it is shown. ([#4659](https://github.com/transmission/transmission/pull/4659))
+* Fixed `4.0.0-beta.1` regression that incorrectly handled corrupt blocklists. ([#4705](https://github.com/transmission/transmission/pull/4705))
+* Increased the font size in the main window's torrent list. ([#4557](https://github.com/transmission/transmission/pull/4557))
+* Fixed [libpsl](https://github.com/rockdaboot/libpsl) dependency build issue on macOS. ([#4642](https://github.com/transmission/transmission/pull/4642))
+
+### Qt Client
+
+* Made display order of  speed limits consistent between Properties, Details dialogs. ([#4677](https://github.com/transmission/transmission/pull/4677))
```
Comment (PriitUring) [inline news/news-4.0.0.md:78]: Typo in line 78: "of speed" <- no need for double space between these 2 words.

File: news/news-4.0.0.md (hunk 2 above)
Comment (ckerr) [reply news/news-4.0.0.md]: Fixed a0305c72fc32a3022463941704e4325830baa7e5

File: news/news-4.0.0.md (hunk 1 above)
Comment (ckerr) [reply news/news-4.0.0.md]: Fixed c5ded48fec22fc581c06f227e8396abf76fd6420

File: news/news-4.0.0.md (hunk 3 above)
Comment (ckerr) [reply news/news-4.0.0.md]: Fixed c5ded48fec22fc581c06f227e8396abf76fd6420

**Pull request merged**